src/CorsService.php

```diff
@@ -358,46 +358,64 @@ protected function isOriginAllowed($origin)
             return true;
         }
 	    
-    	if (preg_match('#\*\.#', implode('|', $this->allowOrigins)) !== 0) {
-			//All subdomains should match !
-			$domains = array();
-			foreach ($this->allowOrigins as $entry) {
-				if (preg_match('#\*\.#', $entry) !== 0) {
-					$domains[] = $entry;
-				}
-			}
-			//Lets parse the uri to extract some parts
-			$parser = new UriParser();
-			$originParsed = $parser->parse($origin);
-
-			foreach ($domains as $domain) {
-				//UriParser consider (rightly) *. as an invalid subdomain, so we have to replace it
-				$domainParsed = $parser->parse(str_replace('*', 'wildcard', $domain));
-				if ($domainParsed['scheme'] == $originParsed['scheme']) {
-					//Same protocol, next step
-					$domainHost = str_replace('wildcard', '*', $domainParsed['host']);
-					/* 
-					 * Create the final RegExp, like (.*\.)?example\.com
-					 * This would match :
-					 *  - example.com
-					 *  - www.example.com
-					 *  - subdomain.example.com
-					 */
-					//Escape all the dots
-					$domainHostRE = str_replace('.', '\.', $domainHost);
-					//Place the RegExp
-					$domainHostRE = str_replace('*\.', '(.+\.)?', $domainHostRE);
-					
-					if (preg_match('#'.$domainHostRE.'#', $originParsed['host']) !== 0) {
-						return true;
-					}
-				}
-			}
-		}
+    	if ($this->hasWildcardOrigins()) {
+	    if ($this->isOriginInWildcardOrigins($origin)) {
+	        return true;
+	    }
+        }
```

Comment:
Nit: align indents.

Tip: use an editor on your computer rather than the GitHub editor.

Reply:
I'll try to use git correctly instead of copy-pasting, yes...